lightning-background-processor/src/lib.rs

```diff
@@ -158,13 +158,27 @@ impl BackgroundProcessor {
 				}
 			}
 		});
-		Self { stop_thread: stop_thread_clone, thread_handle: handle }
+		Self { stop_thread: stop_thread_clone, thread_handle: Some(handle) }
 	}
 
 	/// Stop `BackgroundProcessor`'s thread.
-	pub fn stop(self) -> Result<(), std::io::Error> {
+	pub fn stop(mut self) -> Result<(), std::io::Error> {
+		assert!(self.thread_handle.is_some());
```

Comment:
Shouldn't be able to hit this one because it takes ownership of `self` here, but fair on the other one.

Reply:
`join` also takes ownership. More generally, once the thread is joined `BackgroundProcessor` is no longer useful. It needs to be recreated with `start`. The only reason we need to use an `Option` is because `drop` takes a reference.